lib/jdomodels/src/test/java/org/sagebionetworks/repo/model/dbo/auth/LoginLockoutStatusDaoImplTest.java

```diff
@@ -0,0 +1,181 @@
+		//There should be no lock info for this user at this point.
+		// call under test
+		loginLockoutStatusDao.incrementLockoutInfoWithNewTransaction(userOneId);
+		LockoutInfo info = loginLockoutStatusDao.getLockoutInfo(userOneId);
+		assertNotNull(info);
+		assertEquals(1L, info.getNumberOfFailedLoginAttempts());
+		assertNotNull(info.getRemainingMillisecondsToNextLoginAttempt());
+		assertTrue(info.getRemainingMillisecondsToNextLoginAttempt() < (1<<2));
+	}
+	
+	/**
+	 * This test might be sensitive to timing.
+	 * @throws InterruptedException
+	 */
+	@Test
+	public void testIncrementLockoutInfoWithNewTransactionWithMultipleIncrements() throws InterruptedException {
+		long startMS = 0L;
+		// 10 attempts should lockout for 1024 MS.
+		long numberOfAttempts = 10L;
+		for(long i=0; i<numberOfAttempts; i++) {
+			// call under test
+			startMS = System.currentTimeMillis();
+			loginLockoutStatusDao.incrementLockoutInfoWithNewTransaction(userOneId);
+		}
+		LockoutInfo info = loginLockoutStatusDao.getLockoutInfo(userOneId);
+		// This helps adjust for slow running test machines.
+		long millisecondsSinceLastIncrement = System.currentTimeMillis()-startMS;
+		assertNotNull(info);
+		assertEquals(numberOfAttempts, info.getNumberOfFailedLoginAttempts());
+		assertTrue(info.getRemainingMillisecondsToNextLoginAttempt() > (1<<numberOfAttempts-1)-millisecondsSinceLastIncrement);
+		assertTrue(info.getRemainingMillisecondsToNextLoginAttempt() < (1<<numberOfAttempts+1));
+		// wait for the lock to expire if it is not already.
+		if(info.getRemainingMillisecondsToNextLoginAttempt()>0) {
+			Thread.sleep(info.getRemainingMillisecondsToNextLoginAttempt()+10L);
+		}
+		info = loginLockoutStatusDao.getLockoutInfo(userOneId);
+		assertNotNull(info);
+		assertEquals(numberOfAttempts, info.getNumberOfFailedLoginAttempts());
+		assertTrue(info.getRemainingMillisecondsToNextLoginAttempt() <= 0L);
+	}
+	
+	@Test
+	public void testResetLockoutInfoWithNewTransactionWithNoData() {
+		// call under test
+		loginLockoutStatusDao.resetLockoutInfoWithNewTransaction(userOneId);
+		LockoutInfo info = loginLockoutStatusDao.getLockoutInfo(userOneId);
+		assertNotNull(info);
+		assertEquals(0L, info.getNumberOfFailedLoginAttempts());
+		assertTrue(info.getRemainingMillisecondsToNextLoginAttempt() < 0L);
+	}
+	
+	@Test
+	public void testResetLockoutInfoWithNewTransactionWithData() {
+		loginLockoutStatusDao.incrementLockoutInfoWithNewTransaction(userOneId);
+		LockoutInfo info = loginLockoutStatusDao.getLockoutInfo(userOneId);
+		assertNotNull(info);
+		assertEquals(1L, info.getNumberOfFailedLoginAttempts());
+		// call under test
+		loginLockoutStatusDao.resetLockoutInfoWithNewTransaction(userOneId);
+		info = loginLockoutStatusDao.getLockoutInfo(userOneId);
+		assertNotNull(info);
+		assertEquals(0L, info.getNumberOfFailedLoginAttempts());
+		assertTrue(info.getRemainingMillisecondsToNextLoginAttempt() < 0L);
```

Comment:
We should probably protect ourselves from potential really fast tests "< 0L" -> "<= 0L"